common/transport/mqtt/src/main/java/org/thingsboard/server/transport/mqtt/session/DeviceSessionCtx.java

```diff
@@ -255,4 +255,8 @@ public Collection<MqttMessage> getMsgQueueSnapshot(){
         return Collections.unmodifiableCollection(msgQueue);
     }
 
+    public boolean isSparkplug () {
+        return ((MqttDeviceProfileTransportConfiguration) this.getDeviceProfile().getProfileData().getTransportConfiguration()).isSparkPlug();
```

Comment:
Are you sure that we have here MqttDeviceProfileTransportConfiguration? What if we have DefaultDeviceProfileTransportConfiguration here?

Reply:
@ashvayka